src/python/pants/engine/scheduler.py

```diff
@@ -37,6 +37,8 @@
 
 logger = logging.getLogger(__name__)
 
+WORKUNIT_TY = Dict[str, Any]
```

Comment:
What does TY mean? Also, normally with MyPy type aliases, you use PascalCase.

Reply:
It's likely worth it to use a `TypedDict` here: https://www.python.org/dev/peps/pep-0589/#class-based-syntax.

`TypedDict`s are helpful when you know the exact shape of your dictionary. They allow MyPy to know, for example, that this dict will have exactly two keys named `started` and `completed`, whereas now it's an arbitrary dictionary with 0-infinity keys.

You'll need to import `from typing_extensions import TypedDict` and check if `3rdparty/python:typing-extensions` is in the BUILD file.